### PR TITLE
KOGITO-6300: DMN Runner Table Improvements

### PR DESCRIPTION
**Pull request by ljmotta**
## JIRA
https://issues.redhat.com/browse/KOGITO-6301
https://issues.redhat.com/browse/KOGITO-6302

## On This PR
- Refactor the Grid class to be a React hook.
- Fix changing a Decision Node type breaking the table.
- Disable the DMN Runner Table on `bpmn` and `pmml` files.
- Remove unnecessary files.

### CODE REVIEW

File: packages/unitables/src/core/Grid.tsx [hunk 1]
```diff
@@ -148,116 +125,300 @@ export class Grid {
...
+    (
+      outputTypeMap: Map<string, OutputTypesFields>,
+      properties: DmnSchemaProperties[],
+      jsonSchemaBridge: DmnTableJsonSchemaBridge
+    ) => {
+      return Object.entries(properties).map(([name, property]: [string, DmnSchemaProperties]) => {
+        if (property["x-dmn-type"]) {
+          const dataType = getDataTypeProps(property["x-dmn-type"]).dataType;
+          outputTypeMap.set(name, { type: property.type, dataType, name });
+          return { name, type: property.type, width: 150, dataType };
```
Comment (tiagobento) [inline packages/unitables/src/core/Grid.tsx]: Is `150` the "minimum size"? If so, I think we could create a constant for it. There's other places using the same `150` value. Are they related?

**Review by tiagobento — APPROVED** [review]

**Comment by jomarko** [comment]
@ljmotta may I ask you to link this PR with a jira?

File: packages/unitables/src/core/Grid.tsx (hunk 1 above)
Comment (ljmotta) [reply packages/unitables/src/core/Grid.tsx]: Yes, already extract to a constant

**Comment by ljmotta** [comment]
@jomarko I´ve added the JIRAS in the description

**Comment by tiagobento** [comment]
@jomarko Hi Jozef, can you please take a look at this PR?

**Review by jomarko — COMMENTED** [review]
@ljmotta hello, thank you for the PR.

I would like to mention an issue found during review, that I reported a separately, because it requires changes in kogito-editors-java and probably a editor version bump to be propagated into kogito-tooling? 
- https://issues.redhat.com/browse/KOGITO-6355

I would like also add some test before we proceed with merge of non trivial changes. I prepared a test here, please review and merge, so we run CI with them. locally they pass.
- https://github.com/ljmotta/kogito-tooling/pull/9

**Comment by ljmotta** [comment]
@jomarko Hi, thanks for taking your time reviwing this PR and making Cypress test for it.

**Comment by tiagobento** [comment]
@jomarko Is this one ready to be merged?

**Review by jomarko — COMMENTED** [review]
Reading the code, I think this PR aims to improve switch from original runner view to table view and vice versa. 

I think I found these issues.

### Output columns width
I had table with three rows
Screenshot from 2021-12-06 10-07-12

Then clicked *Run* to hide it, and again *Run* to display it. Notice output columns width
Screenshot from 2021-12-06 10-07-26

### Update row and click view other row
I updated second row
Screenshot from 2021-12-06 10-10-16

Immediately click a button to see row 3
Screenshot from 2021-12-06 10-10-30

Then tried to see row 2 details, but got wrong data
Screenshot from 2021-12-06 10-10-37

**Comment by ljmotta** [comment]
@jomarko Thanks for your review. Those two new issues you found are not in the scope of this PR. Could you please create two separete JIRAs? I'll address it in another PR.

**Comment by jomarko** [comment]
@ljmotta thank you for your comment.

I reported issues mentioned above https://github.com/kiegroup/kogito-tooling/pull/692#pullrequestreview-823699544 as:
- https://issues.redhat.com/browse/KOGITO-6382
- https://issues.redhat.com/browse/KOGITO-6383

**Review by jomarko — APPROVED** [review]

**Pull request merged**